src/main/java/capstone/mfslbackend/service/FantasyLeagueService.java

```diff
@@ -145,35 +148,47 @@ public List<FantasyWeek> createFantasyLeagueSchedule(Long leagueId) {
             schedule.add(new ArrayList<>(matches));
 
         }
-        int w = 0;
-        // Create the weeks
-        for (int matchIndex = 0; matchIndex < matches.size(); matchIndex += 2) {
-            FantasyWeek fantasyWeek = new FantasyWeek();
-            FantasyTeam teamA = matches.get(matchIndex);
-            FantasyTeam teamB = matches.get(matchIndex + 1);
-
-
-            Set<FantasyWeek> weeksA = teamA.getFantasyWeeks();
-            Set<FantasyWeek> weeksB = teamB.getFantasyWeeks();
-            fantasyWeek.setFantasyTeamA(teamA);
-            fantasyWeek.setFantasyTeamB(teamB);
-            weeksA.add(fantasyWeek);
-            weeksB.add(fantasyWeek);
-            teamA.setFantasyWeeks(weeksA);
-            teamB.setFantasyWeeks(weeksB);
-
-            //fix order number once decided
-            teamA.setOrderNumber(matchIndex);
-            teamB.setOrderNumber(matchIndex + 1);
-
-            fantasyWeekRepository.save(fantasyWeek);
+
+        while (fantasyWeeks.size() < 38) {
+            for (List<FantasyTeam> weekMatches : schedule) {
+                if (fantasyWeeks.size() >= 38) {
+                    break;
+                }
+                for (int matchIndex = 0; matchIndex < weekMatches.size(); matchIndex += 2) {
+                    FantasyTeam teamA = weekMatches.get(matchIndex);
+                    FantasyTeam teamB = weekMatches.get(matchIndex + 1);
+
+                    FantasyWeek fantasyWeek = new FantasyWeek();
+                    fantasyWeek.setFantasyTeamA(teamA);
+                    fantasyWeek.setFantasyTeamB(teamB);
+                    fantasyWeek.setWeekNumber(fantasyWeeks.size() + 1);
+
+                    LocalDate startDate = LocalDate.now().plusWeeks(fantasyWeeks.size());
```

Comment:
this won't work because there can be more than one fantasy week for a calendar week(me vs you, jeet vs roosh)